src/ResourceManager/ContainerRegistry/Commands.ContainerRegistry/Commands/NewAzureContainerRegistry.cs

```diff
@@ -0,0 +1,110 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections;
+using System.Management.Automation;
+using Microsoft.Azure.Management.ContainerRegistry.Models;
+using Microsoft.Azure.Management.ResourceManager.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using DeploymentState = Microsoft.Azure.Commands.ResourceManager.Cmdlets.Entities.ProvisioningState;
+
+namespace Microsoft.Azure.Commands.ContainerRegistry
+{
+    [Cmdlet(VerbsCommon.New, ContainerRegistryNoun, SupportsShouldProcess = true),
+        OutputType(typeof(PSContainerRegistry))]
+    public class NewAzureContainerRegistry : ContainerRegistryCmdletBase
+    {
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Resource Group Name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Container Registry Name.")]
+        [Alias(ContainerRegistryNameAlias, RegistryNameAlias, ResourceNameAlias)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Container Registry SKU. " + AllowedSkuNames)]
+        [Alias(ContainerRegistrySkuAlias, RegistrySkuAlias)]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(SkuTier.Basic, IgnoreCase = false)]
+        public string Sku { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Container Registry Location. Default to the location of the resource group.")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Indicates whether the admin user is enabled.")]
+        [ValidateNotNull]
+        [Alias(AdminEnabledAlias)]
+        public bool? AdminUserEnabled { get; set; }
```

Comment:
@djyou we talked to the team and decided the following:

- `New-*` will have a SwitchParameter for `EnableAdminUser`
- `Update-*` will have two SwitchParameters for `EnableAdminUser` and `DisableAdminUser` that are in separate parameter sets

Let me know if you have any questions about this design